src/libs/ajax.js

```diff
@@ -1122,7 +1122,8 @@ const Dockstore = signal => ({
 
 const Martha = signal => ({
   getDataObjectMetadata: async url => {
-    const res = await fetchMartha('martha_v2', _.mergeAll([jsonBody({ url }), appIdentifier, { signal, method: 'POST' }]))
+    // TODO: switch to 'martha_v3' to point to updated endpoint
+    const res = await fetchMartha('martha_v2', _.mergeAll([jsonBody({ url }), authOpts(), appIdentifier, { signal, method: 'POST' }]))
```

Comment:
This will send auth headers to martha, which will cause it to call bond. I don't think we want that (for v2 or v3), since we're not looking to use any bond SA from this call. (The SA gets looked up and used when generating a signed URL separately)

Reply:
`martha_v2` only calls Bond for certain providers (see https://github.com/broadinstitute/martha/blob/dev/common/bond.js#L44) and `martha_v3` behaves the same way; it is not based on whether auth is sent

Reply:
Some of the new repos supported by `martha_v3` require `auth` to retrieve metadata. Instead of relying on clients like `terra-ui` to parse the URI and then figure out if the URI requires `auth` for metadata, `martha_v3` just always requires auth and then internally, based on the host, contacts bond.

There is an ongoing discussion regarding enabling "partial responses" by selecting only certain fields to include, such as `googleServiceAccount`. TOL: it would also be nice to perform the URL pre-signing in a single request, especially as Terra begins to support DOS/DRS servers that do not resolve to GCS. For example, the AnVIL returns `s3://` and `https://` but not `gs://`.

Reply:
FWIW from line https://github.com/broadinstitute/martha/blob/dev/common/bond.js#L45 the bond call is also conditional on whether the auth header is present. This may be a benign change, and in any case it's not intended to be  live for very long. (Thanks for the clarification on v3 needing auth, that makes sense)

Regarding 'partial responses', rather than trying to bundle multiple operations together and then provide additional mechanisms to separate them, we could instead provide smaller, single purpose building blocks and let downstream consumers compose them according to their needs. We ran into an issue previously with `fileSummaryV1` where we had to abandon it because it was trying to do too much, and didn't interact well with requester pays. Composing operations together at the point-of-use is much easier than taking a pre-bundled thing and trying to pull it apart.